### ISSUE TITLE
Screwdriver not as a item

### ISSUE THREAD

**Issue by TekhnaeRaav** [minetest/minetest#6901]
The screwdriver ability should not be an item that you have to craft, this is something you should be able to do any time you want.
Reasons? Because much of the reason you need a screwdriver is because you can't place blocks how you want to in the first place, like putting a slab on the top half of a block instead of the bottom half of one. I believe this should have to be activated by a key, not an item that you have to get by mining **iron**.

**Comment by benrob0329** [comment]
I believe better placement code would solve this (and the placement code has already improved **a lot** since just last year) and get rid of the need for a screwdriver altogether.

**Comment by rubenwardy** [comment]
Minetest game issue

**Comment by benrob0329** [comment]
I'm not so sure, the rotation of nodes is a core engine feature.

**Comment by TekhnaeRaav** [comment]
Yes, I forgot to mention why I put my issue _here._ Because if to make it a button and a default function in minetest, it would probably need an engine change.
Unless you want to _decide_ this on minetest game first.

**Comment by MarkuBu** [comment]
Placing slabs on the top half of a node is possible since 0.4.16. 

Same for stairs

**Comment by jastevenson303** [comment]
A key for rotation, making the screwdriver optional, is a great idea.

**Comment by 4w** [comment]
Yes, that would be great because it moves another key feature from a subgame to the core.

**Comment by MarkuBu** [comment]
I don't really see the issue here. For some nodes like slabs and stairs I would add better auto replace and auto rotate functions. But they depend on the shape of the node, so each mod needs to add these functions for the specific nodes. And when I mentioned that I was told that more complex functions are not welcome.

And I don't see a problem to use an item to rotate nodes. In creative mode it doesn't matter because you get it for free. And in survival it is one of the few progression elements in MTG

**Comment by jastevenson303** [comment]
It's nice to have a screwdriver AND a key, and depending on the game either
is optional.

**Comment by ThomasMonroe314** [comment]
I agree that a key would be awesome to have, however that said, the idea may have to be tweaked a little first.
for example, @MarkuBu said:
> And I don't see a problem to use an item to rotate nodes. In creative mode it doesn't matter because you get it for free. And in survival it is one of the few progression elements in MTG

you could make it behave similar to other features and only be enabled with a certain priv.

**Comment by TekhnaeRaav** [comment]
>Placing slabs on the top half of a node is possible since 0.4.16.

Great! I must have last tested this on 0.4.15.

>And I don't see a problem to use an item to rotate nodes. In creative mode it doesn't matter because you get it for free. And in survival it is one of the few progression elements in MTG

But like I said if your building you really should have been able to put the block where you wanted.
    Although I do think a priv could work, because I hate to have the screwdriver taking up the hotbar. That way you only need it to be in your inventory.

**Comment by 4w** [comment]
> Also, the block placement is actually pretty good now in minetest 

Except for wooden logs.

**Comment by TekhnaeRaav** [comment]
Oh, your right it does have some needed improvements.

**Comment by paramat** [comment]
Screwdriver (or rather, ability to rotate to any orientation) will always be needed, and it does seem like something that should be in builtin or engine, not sure about how to implement.

**Comment by TekhnaeRaav** [comment]
Yes, screwdriver will always be needed, I was just pointing out why this ability should not be an item. :+1:

**Comment by paramat** [comment]
Yes it seems to me that the ability to place a node in any orientation should always be enabled and not dependent on an item that needs crafting.
This may have to wait for custom key bindings if done with controls.
Or alternatively (or until then) in a game the screwdriver could always be given to players in survival mode, using 'give initial stuff' in MTG.

**Comment by 4w** [comment]
> Or alternatively (or until then) in a game the item could always be given to players in survival mode, using the 'give initial stuff' API in MTG.

\So you need to depend your subgame on another subgame to get API access? AM I the only one who feels that this is completely wrong?

**Comment by paramat** [comment]
Uh i was unclear there, fixed, a game using another game's API is of course not possible. I prefer an engine feature.

**Comment by Pantyhose-X** [comment]
> Uh i was unclear there, fixed, a game using another game's API is of course not possible.
> I would prefer an engine feature.

The ability to rotate to any direction node, which seems like it should be built into the engine
* https://content.minetest.net/packages/entuland/rhotator/
* https://content.minetest.net/packages/12Me21/screwdriver2/
* https://content.minetest.net/packages/niwla23/batch_screwdriver/
